march_hardware_builder/src/hardware_builder.cpp

```diff
@@ -257,3 +269,24 @@ void HardwareBuilder::initUrdf()
     this->init_urdf_ = false;
   }
 }
+
+std::string rightHandJustifyString(std::string input)
+{
+  std::string delimiter = "\n";
+  if (input.length() > delimiter.length())
+  {
+    while (input.size() < 4 + delimiter.length())  // check until the 16-bit number has been filled
+    {
+      input.insert(0, "0");
+    }
+  }
+  return input;
+}
```

Comment:
Is there a reason that is method is not being used?

Reply:
I have previously used an iterating function that made use of this one. The current implementation used a type of copying to move from ifstream to stringstream. I forgot to add this function to the converter.

Reply:
I have adjusted the code so that it is used again, but the use case for the function has expanded to strings containing all the data of the file.